### ISSUE TITLE
scrollreveal.js:371 Uncaught TypeError: Cannot read property 'config' of undefined - Bootstrap 4 Compatibility Error

### ISSUE THREAD

**Issue by drago2308** [jlmakes/scrollreveal.js#257]
Hi,

I am using the windows operating system with the CMS Concrete 5.8 with a custom package of bootstrap 4.
I am using the latest version of chrome and firefox developers edition to test the website.
I am using the latest version of the scrollreveal plugin.
The issue arises when the compiled version of bootstrap 4 is implemented - at least i think that is causing the issue.

scrollreveal.js:371 Uncaught TypeError: Cannot read property 'config' of undefined which points to this part of the code.

```js
function _configure(elem, config) {

            // If the element hasn’t already been configured, let’s use a clone of the
            // defaults extended by the configuration passed as the second argument.
            if (!elem.config) {
                elem.config = sr.tools.extendClone(sr.defaults, config);
            }

            // Otherwise, let’s use a clone of the existing element configuration extended
            // by the configuration passed as the second argument.
            else {
                elem.config = sr.tools.extendClone(elem.config, config);
            }

            // Infer CSS Transform axis from origin string.
            if (elem.config.origin === 'top' || elem.config.origin === 'bottom') {
                elem.config.axis = 'Y';
            } else {
                elem.config.axis = 'X';
            }

            // Let’s make sure our our pixel distances are negative for top and left.
            // e.g. config.origin = 'top' and config.distance = '25px' starts at `top: -25px` in CSS.
            if (elem.config.origin === 'top' || elem.config.origin === 'left') {
                elem.config.distance = '-' + elem.config.distance;
            }
        }
```
Sorry if this issue is just my setup.

Thanks!

**Comment by jlmakes** [comment]
Hey there @drago2308, from what I can tell so far...

The error means that `elem` is undefined, which sounds like an element ID that is stored in ScrollReveal (via the `data-sr-id` attributes it attaches to elements) is not actually correctly referencing an element inside ScrollReveal’s store.

Maybe something is happening with the DOM, or the ScrollReveal instance, causing them to be out of sync—but I’ll know more if you can:

-  Provide a little more of your JavaScript—for example, where are you instantiating ScrollReveal, and what elements/selectors are you passing to `sr.reveal()`?

- Link to an JSBin or any live code I can inspect.

**Comment by drago2308** [comment]
Ah I'm so sorry to cause you an annoyance.

Something (I don't know what) must have been broken in the DOM when I installed fro the first time, However I reinstalled concrete and then loaded the plugin the same way as I did previously and it worked fine this time, so it must have been an odd error with the concrete 5 CMS (not bootstrap 4, not your error at all, so again sorry).

For extra unnecessary information: 
The way I called the 'sr.reveal()' was trough a custom plugin that searched the page to see if a dom element first had the class with the name of somethings set in an array, if a match then link it with the names custom animation, this way there was no way that their could be an error in mismatching. Hence why I concluded that it wasn't my code selecting the wrong DOM element and connecting it with the wrong animation. (turns out this error was nothing anyway)

Thanks for telling me what the error means in general, It really will help me if the error comes up again or something similar. 

Sorry again for reporting an issue about something that really wasn't.
